### PR TITLE
GCE: Separate the network's project from the rest of the project

### LINKED ISSUE #51846: GCE: Cluster turn-up with a network that belongs to another project
/kind feature
/sig network
/area platform/gce

### PR DESCRIPTION
**Pull request by nicksardo**
**What this PR does / why we need it**:
PR allows the user to specify a different project for network resources during cluster turn-up. 

Depends on #51725
Fixes #51846
/assign @bowei

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by bowei** [comment]
is there an issue for XPN support?

**Comment by nicksardo** [comment]
Still testing this PR.  No open issue for XPN.

**Comment by nicksardo** [comment]
Please review https://github.com/kubernetes/kubernetes/pull/51739/commits/ba7e20bd9ee70ffe72aadf0d04c0838938700056.

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -499,28 +502,30 @@ function create-firewall-rule() {
...
-    ret="${ret},subnet=${alias_subnetwork}"
+    ret="${ret},subnet=${subnetURL}"
     ret="${ret},aliases=pods-default:${alias_size}"
     ret="${ret} --no-can-ip-forward"
   else
-    if [[ ${ENABLE_BIG_CLUSTER_SUBNETS} != "true" || (${PREEXISTING_NETWORK} = "true" && "${PREEXISTING_NETWORK_MODE}" != "custom") ]]; then
-      ret="--network ${network}"
-    else
-      ret="--subnet=${network}"
-    fi
```
Comment (nicksardo) [inline cluster/gce/util.sh]: Since we retrieve the subnet in `create-subnets`, I believe we can always specify the subnet regardless of the situation.

**Comment by nicksardo** [comment]
Could I get more eyes on this change? There's a possibility that I missed a "PROJECT" -> "NETWORK_PROJECT" change. Also, the `make-gcloud-network-arguments` could be verified acceptable. 

If you guys have time, PTAL.
/cc @shyamjvs @MrHohn 


/assign @mikedanese

File: cluster/common.sh [hunk 2]
```diff
@@ -610,11 +610,13 @@ ENV_TIMESTAMP: $(yaml-quote $(date -u +%Y-%m-%dT%T%z))
 INSTANCE_PREFIX: $(yaml-quote ${INSTANCE_PREFIX})
 NODE_INSTANCE_PREFIX: $(yaml-quote ${NODE_INSTANCE_PREFIX})
 NODE_TAGS: $(yaml-quote ${NODE_TAGS:-})
-NODE_NETWORK: $(yaml-quote ${NODE_NETWORK:-})
-NODE_SUBNETWORK: $(yaml-quote ${NODE_SUBNETWORK:-})
+NODE_NETWORK: $(yaml-quote ${NETWORK:-})
+NODE_SUBNETWORK: $(yaml-quote ${SUBNETWORK:-})
```
Comment (MrHohn) [inline cluster/common.sh]: Curious: Is it safe to rename these variables?

Comment (MrHohn) [reply cluster/common.sh]: Ah never mind, just noticed that they are being added in #51725.

File: cluster/gce/util.sh [hunk 3]
```diff
@@ -798,25 +806,26 @@ function create-network() {
...
-    --project "${PROJECT}" \
+    --project "${NETWORK_PROJECT}" \
     --prefix-length=19 \
     --quiet
 }
 
 
 # Vars set:
+#   SUBNETWORK
 #   NODE_SUBNETWORK
```
Comment (MrHohn) [inline cluster/gce/util.sh]: Remove this?

Comment (nicksardo) [reply cluster/gce/util.sh]: Good catch, thanks.

File: cluster/gce/util.sh [hunk 4]
```diff
@@ -1316,13 +1340,16 @@ function create-nodes() {
...
+# - SUBNETWORK
 # - IP_ALIAS_SIZE
 function create-heapster-node() {
   local network=$(make-gcloud-network-argument \
-      "${NETWORK}" "" \
+      "${NETWORK_PROJECT}" \
+      "${REGION}" \
+      "${NETWORK}"
+      "${SUBNETWORK}" \
+      "" \ # Address
```
Comment (bowei) [inline cluster/gce/util.sh]: can't do this with the comment

Comment (nicksardo) [reply cluster/gce/util.sh]: Good catch, thanks!

**Comment by nicksardo** [comment]
Rebased
/cc @bowei

**Comment by dims** [comment]
Reapplying lgtm as this PR had to be rebased (original lgtm from @bowei)

/lgtm

**Pull request merged**